packages/bot-web-ui/src/pages/server-side-bot/add-bot/add-bot.tsx

```diff
@@ -0,0 +1,241 @@
+            );
+            Object.keys(data).forEach(key => {
+                initial_value[key as keyof TFormData] = data[key];
+                setValue(key, data[key]);
+            });
+        } catch {
+            data = null;
+        }
+        return data;
+    };
+
+    const getInitialValue = () => {
+        const data = getSavedValues();
+        initial_value = {
+            name: data?.name ?? 'Martingale',
+            symbol: data?.symbol ?? qs_config.QUICK_STRATEGY.DEFAULT.symbol,
+            tradetype: data?.tradetype ?? '',
+            type: data?.type ?? '',
+            durationtype: data?.durationtype ?? qs_config.QUICK_STRATEGY.DEFAULT.durationtype,
+            duration: data?.duration ?? '1',
+            stake: data?.stake ?? '1',
+            loss: data?.loss ?? '',
+            profit: data?.profit ?? '',
+            size: data?.size ?? String(qs_config.QUICK_STRATEGY.DEFAULT.size),
+        };
+        return initial_value;
+    };
+
+    React.useEffect(() => {
+        return () => {
+            is_mounted.current = false;
+        };
+    }, []);
+
+    React.useEffect(() => {
+        initializeLossThresholdWarningData();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const getErrors = (formikData: TFormData) => {
+        const sub_schema: Record<string, any> = {};
+        config.forEach(group => {
+            if (!group?.length) return null;
+            group.forEach(field => {
+                if (field?.validation?.length && field?.name) {
+                    if (field.validation.includes('number')) {
+                        let schema: Record<string, any> = Yup.number().typeError(localize('Must be a number'));
+                        let min = 0;
+                        let max = 10;
+                        let min_error = getErrorMessage('MIN', min);
+                        let max_error = getErrorMessage('MAX', max);
+                        let integer_error_message = '';
+                        if (field.name === 'duration' && current_duration_min_max) {
+                            min = current_duration_min_max.min;
+                            max = current_duration_min_max.max;
+                            min_error = getErrorMessage('MIN', min, 'DURATION');
+                            max_error = getErrorMessage('MAX', max, 'DURATION');
+                        }
+                        const should_validate = field.should_have
+                            ? field.should_have?.every(item => {
+                                  const item_value = formikData?.[item.key]?.toString();
+                                  if (item.multiple) return item.multiple.includes(item_value);
+                                  return formikData?.[item.key] === item.value;
+                              })
+                            : true;
+                        if (should_validate && field.name === 'max_stake') {
```

Comment:
there are a few if statements controlling the min, max, error etc - are these specific checks? if it is specific then there should be an escape statement out of the current iteration. because if all of these checks are true - it will overwrite each other based on the latest check

Reply:
Yes @yashim-deriv indeed, this is not permanent fix, we will come back to this and refactor later on

Reply:
sure, no issue @sandeep-deriv